### DISCUSSION TITLE
Authenticate private Terraform repository with Spacelift Registry

### DISCUSSION THREAD

**Discussion by tommyallieri** [renovatebot/renovate#25259] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.26

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We host Terraform module in a private registry with Spacelift. We want to use the hostRules to handle authentication and successfully authenticate for Renovate to see the different versions for each module.

### What you tried so far.

Terraform module example
```
module "blah" {
  source  = "spacelift.io/<org>/<module-name>/default"
  version = "x.y.z"
```

We initially didn't have any authentication and received this error
```
Failed to look up terraform-module package spacelift.io/<org>/<module-name>/default
```

We then created a Spacelift API key and added this config
```
         "hostRules": [
           {
             "hostType": "terraform-module",
             "matchHost": "app.spacelift.io",
             "token": "***********"
           }
         ],
```
We can see authn is successful (debug logs show 200 and Spacelift UI confirms login), but we still get the same error `Failed to look up terraform-module package spacelift.io/<org>/<module-name>/default`

matchHost set to `spacelift.io` also successful logs in, but same error at the end

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Failed to look up terraform-module package spacelift.io/<org>/<module-name>/default (repository=<repo-name>, packageFile=main.tf, dependency=spacelift.io/<org>/<module-name>/default)
           {
             "deps": [
               {
                 "depName": "spacelift.io/<org>/<module-name>/default",
                 "depType": "module",
                 "currentValue": "x.y.z",
                 "registryUrls": ["https://spacelift.io"],
                 "datasource": "terraform-module",
                 "updates": [],
                 "packageName": "spacelift.io/<org>/<module-name>/default",
                 "versioning": "hashicorp",
                 "warnings": [
                   {
                     "topic": "spacelift.io/<org>/<module-name>/default",
                     "message": "Failed to look up terraform-module package spacelift.io/<org>/<module-name>/default"
                   }
                 ]
               },

[...]

DEBUG: http statistics (repository=<repo-name>)
       "urls": {
         "https://app.spacelift.io/registry/modules/v1/<org>/<module-name>/default/versions (GET,200)": 1,
         "https://spacelift.io/.well-known/terraform.json (GET,200)": 1
       },
       "hostStats": {
         "api.bitbucket.org": {
           "requestCount": 7,
           "requestAvgMs": 1388,
           "queueAvgMs": 0
         },
         "app.spacelift.io": {
           "requestCount": 3,
           "requestAvgMs": 1039,
           "queueAvgMs": 0
         },
         "registry.terraform.io": {
           "requestCount": 7,
           "requestAvgMs": 295,
           "queueAvgMs": 0
         },
         "spacelift.io": {"requestCount": 1, "requestAvgMs": 1217, "queueAvgMs": 0}
       },
```

</details>

**Comment by rarkins** [comment]
If the http request looks ok then perhaps there's something about the response contents which Renovate doesn't like. Can you `curl` or similar the relevant URL and copy paste the JSON contents here?

**Reply by tommyallieri** [reply]
thanks @rarkins. I got in touch with the Spacelift team. It seems the problem was with API key permission in the end and Renovate works as expected with a few more permissions.

For the next person with the same problem - simply giving read access to the space with the modules (spacelift jargon) wasn't sufficient, providing read access to the root module via login policy fixed my problem
